src/app/components/elements/EventBookingForm.tsx

```diff
@@ -14,6 +14,112 @@ interface EventBookingFormProps {
   updateAdditionalInformation: (update: AdditionalInformation) => void;
 }
 
+const SchoolYearGroup = ({
+  event,
+  additionalInformation,
+  updateAdditionalInformation,
+}: {
+  event: AugmentedEvent;
+  additionalInformation: AdditionalInformation;
+  updateAdditionalInformation: (update: AdditionalInformation) => void;
+}) => {
+  return (
+    <>
+      <RS.Label htmlFor="year-group" className="form-required">
+        School year group
+      </RS.Label>
+      <RS.Input
+        type="select"
+        id="year-group"
+        name="year-group"
+        value={additionalInformation.yearGroup || ""}
+        onChange={(event) => updateAdditionalInformation({ yearGroup: event.target.value })}
+      >
+        <option value="" />
+        {event.isAStudentEvent && (
+          <>
+            <option value="9">Year 9</option>
+            <option value="10">Year 10</option>
+            <option value="11">Year 11</option>
+            <option value="12">Year 12</option>
+            <option value="13">Year 13</option>
+          </>
+        )}
+        {!event.isStudentOnly && (
+          <>
+            <option value="TEACHER">N/A - Teacher</option>
+            <option value="OTHER">N/A - Other</option>
+          </>
+        )}
+      </RS.Input>
+      {event.isStudentOnly && (
+        <div className="text-muted" data-testid="student-only">
+          {studentOnlyEventMessage(event.id)}
+        </div>
+      )}
+      {(event.isATeacherEvent || additionalInformation.yearGroup == "TEACHER") && (
+        <div className="mt-2 text-right">
+          <a href="/pages/teacher_accounts" target="_blank">
+            Click to upgrade to a teacher account
+          </a>{" "}
+          for free!
+        </div>
+      )}
+    </>
+  );
+};
+
+export const AccessibilityAndMedicalRequirements = ({
+  additionalInformation,
+  updateAdditionalInformation,
+}: {
+  additionalInformation: AdditionalInformation;
+  updateAdditionalInformation: (update: AdditionalInformation) => void;
+}) => {
+  const data = [
+    {
+      label: "Accessibility requirements",
+      id: "access",
+      popover:
+        "For example, please let us know if you need wheelchair access, hearing loop or if we can help with any special adjustments.",
+      type: "accessibilityRequirements",
+    },
+    {
+      label: "Dietary requirements or relevant medical conditions",
+      id: "medical",
+      popover:
```

Comment:
The popover is the same for both, I think this is a mistake introduced with the refactoring.
While it's good that you're trying to improve the codebase, this highlights one of the risks of refactoring untested code, even if the intentions were good. One way to avoid this in the future is to write the tests before you do the refactor. That way you can test that the output is the same before and after the refactor. I also think it's probably a good idea to try not to refactor too much in one go, as we don't have an extensive test suite to give us enough confidence that the changes won't break existing functionality.